pip_api/_parse_requirements.py

```diff
@@ -187,15 +461,20 @@ def parse_requirements(
             req: Optional[Union[requirements.Requirement, UnparsedRequirement]] = None
             known, _ = parser.parse_known_args(line.strip().split())
             if known.req:
-                # We want to stop combining strings before an @ sign with space on either side
-                req_str = str()
-                for r in known.req:
-                    if r == "@":
-                        break
-                    req_str += r
+                req_str = str().join(known.req)
+                parsed_req_str = _parse_requirement_url(req_str)
```

Comment:
I'm thinking I should just raise from this function instead of doing the `_check_invalid_requirement` after the fact. I'm guessing the reason we do it this way down below is because we're calling into the `Requirement` constructor which doesn't give much information when it fails.